data-reconciliation-app/src/services/authentication-service.ts

```diff
@@ -127,34 +208,69 @@ export class CertBasedAuthenticationService implements IAuthenticationService {
     }
   }
 
-  // 
-  public isValidIdentity(identityId: string): ServiceResult<boolean> {
-    if (!identityId) {
+  /**
+   * Check if caller's access token is valid
+   * @param {JwtAuthnIdentity} identity JwtAuthnIdentity object
+   * @returns {ServiceResult<boolean>}
+   */
+  public isValidJwtToken(identity: ccfapp.JwtAuthnIdentity): ServiceResult<boolean> {
+
+    if (!identity || !identity.jwt || !identity.jwt.payload || !identity.jwt.payload.sub) {
       return ServiceResult.Failed({
         errorMessage: "Error: invalid caller identity",
         errorType: "AuthenticationError",
       });
     }
 
-    const isMember = this.isActiveMember(identityId);
-    if (isMember.success && isMember.content) {
-      return ServiceResult.Succeeded(true);
+    if (identity.jwt.keyIssuer === "https://demo") { // logic to validate the tokens of demo issuer
```

Comment:
Like Julio's comment - this feels like it should be config/const outside of the AuthService

Do you think we need a AAD-Authentication-Service.ts that implements the specifics of the MSFT version of this so people can plugin other auth providers?

Reply:
I've created a new issue spot this point and added it's link